File: src/AssignmentManagement.Domain/Entities/Assignment.cs
using AssignmentManagement.Domain.Common;
using AssignmentManagement.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AssignmentManagement.Domain.Entities;

public sealed class Assignment : EntityBase
{
    [ForeignKey(nameof(Class))]
    public int ClassId { get; set; }

    public Class? Class { get; set; }

    [Required]
    [MaxLength(255)]
    public string Title { get; set; } = string.Empty;

    public string? Description { get; set; }

    public string? Instructions { get; set; }

    [Required]
    public DateTime DeadlineUtc { get; set; }

    public int MaxMarks { get; set; }

    public AssignmentStatus Status { get; set; }

    public bool AllowResubmission { get; set; }

    public DateTime CreatedAtUtc { get; set; }

    public DateTime UpdatedAtUtc { get; set; }

    public ICollection<Submission> Submissions { get; set; } = [];
}

File: frontend/src/types/assignment.ts
export type AssignmentStatus = "Draft" | "Published" | "Closed";

export interface AssignmentListItem {
  id: number;
  classId: number;
  subjectName: string;
  classLevel: number;
  teacherName?: string;
  title: string;
  description?: string;
  deadlineUtc: string;
  maxMarks: number;
  status: AssignmentStatus;
  allowResubmission: boolean;
  createdAtUtc?: string;
}

export interface AssignmentDetail extends AssignmentListItem {
  instructions?: string;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface CreateAssignmentRequest {
  classId: number;
  title: string;
  description?: string;
  instructions?: string;
  deadlineUtc: string;
  maxMarks: number;
  allowResubmission: boolean;
}

export interface UpdateAssignmentRequest extends CreateAssignmentRequest {}
